lib/src/http_body.dart

```diff
@@ -222,24 +222,24 @@ Future<HttpBody> _process(Stream<List<int>> stream, HttpHeaders headers,
 
```

Comment:
I would refactor this method to move the null check on `contentType` up to here (make it just a check for `headers.contentType == null` instead), and immediately invoke `asBinary`.

Then you can assert non-null in the assignment right after that `var contentType = headers.contentType!;` and remove the other non-null assertions on `contentType`.

Reply:
Actually, it worked when performing the `contentType` null-check after declaring the `asBinary` function. As a result, it's now rewritten as:

```dart
Future<HttpBody> _process(Stream<List<int>> stream, HttpHeaders headers,
    Encoding defaultEncoding) async {
  Future<HttpBody> asBinary() async {
    var builder = await stream.fold<BytesBuilder>(
        BytesBuilder(), (builder, data) => builder..add(data));
    return HttpBody._('binary', builder.takeBytes());
  }

  if (headers.contentType == null) {
    return asBinary();
  }

  var contentType = headers.contentType!;
  //rest of the code...
}
```